api/spec/requests/spree/api/address_books_spec.rb

```diff
@@ -8,6 +8,8 @@ module Spree::Api
     let!(:harry_address_attributes) do
       {
         'name' => 'Harry Potter',
+        'firstname' => 'Harry',
```

Comment:
according to your validation this address record is now invalid. It can either have a name attribute or firstname and lastname. But never both. 

This is just an example for what I saw in lot of your tests. We should make sure we never mix both approaches. This is why I am still in favor of just re-introducing firstname and using name as lastname if firstname is present. But never touch the address (and ignore the firstname) if name is already present.